skills_ml/algorithms/skill_feature_creator/__init__.py

```diff
@@ -0,0 +1,130 @@
+from skills_ml.algorithms.string_cleaners.nlp import NLPTransforms
+from skills_ml.algorithms.skill_feature_creator.structure_features import struct_features
+from skills_ml.algorithms.skill_feature_creator.contextual_features import sent2features, pre_process
+
+from abc import ABC, abstractmethod
+import numpy as np
+import nltk
+
+from functools import reduce
+from itertools import zip_longest
+
+class SequenceFeatureCreator(object):
+    """ Sequence Feature Creator helps users to instantiate different
+    types of feature at once and combine them together into a sentence(sequence) feature array for sequence modeling.
+    It's a generator that outputs a sentence array at a time. A sentence array is composed of word vectors.
+
+    Example:
+        from skills_ml.algorithms.skill_feature_creator import FeatureCreator
+
+        feature_vector_generator = FeatureCreator(job_posting_generator, features=)
+        feature_vector_generator = FeatureCreator(job_posting_generator, features=["StructuralFeature", "EmbeddingFeature"])
+
+    Args:
+        job_posting_generator (generator): job posting generator.
+        sentence_tokenizer (func): sentence tokenization function
+        word_tokenizer (func): word tokenization function
+        features (list): list of feature types ones want to include. If it's None or by default, it includes all the feature types.
+
+    Yield:
+        sentence_array (numpy.array): an array of word vectors represents the words and punctuations in the sentence. The dimension
+                                      is (# of words)*(dimension of the concat word vector)
+    """
+    def __init__(
+            self,
+            job_posting_generator,
+            sentence_tokenizer=NLPTransforms().sentence_tokenize,
+            word_tokenizer=NLPTransforms().word_tokenize,
+            features=None):
+        self.all_features = [ f.__name__ for f in FeatureFactory.__subclasses__()]
+        self.features = features
+        self.job_posting_generator = job_posting_generator
+        self.sentence_tokenizer = sentence_tokenizer
+        self.word_tokenizer = word_tokenizer
+
+    @property
+    def selected_features(self):
+        if self.features is None:
+            return self.all_features
+        elif isinstance(self.features, list):
+            if not (set(self.features) < set(self.all_features)):
+                not_supported = list(set(self.features) - set(self.all_features))
+                raise TypeError("\"{}\" not supported!".format(", ".join(not_supported)))
+            else:
+                return self.features
+        else:
+            raise Exception(TypeError)
+
+    def __iter__(self):
+        feature_objects = [FeatureFactory(self.sentence_tokenizer, self.word_tokenizer)._factory(feature_type) for feature_type in self.selected_features]
```

Comment:
Calling a 'private' (`e.g. _factory`) method from outside is a code smell. I've done it in other places, but I shouldn't have. Should the method just be called `factory`?